ocaml-lsp-server/src/document.mli

```diff
@@ -64,3 +64,7 @@ val close : t -> unit Fiber.t
 val get_impl_intf_counterparts : Uri.t -> Uri.t list
 
 val edit : t -> TextEdit.t -> WorkspaceEdit.t
+
+(** [new_range_on_replace range new_text] returns a new range that will be
+    occupied by [new_text] when [new_text] replaces the text content in [range] *)
+val new_range_on_replace : Range.t -> string -> Range.t
```

Comment:
I find this function hard to understand.  It could be my own fault, but I think we could at least it break it down somewhat. For example:

```ocaml
val extend : Range.t -> Position.t -> Range.t
val end_position : string -> Position.t
```

There could be other ways to do this, so feel free to try something else.

Reply:
I moved the function to `module Range`, renamed, and put more understandable comments. Let me know if it's still unclear.